homeassistant/components/sensor/airvisual.py

```diff
@@ -0,0 +1,304 @@
+    """Merge two dicts into a new dict as a shallow copy."""
+    final = dict1.copy()
+    final.update(dict2)
+    return final
+
+
+class AirVisualBaseSensor(Entity):
+    """Define a base class for all of our sensors."""
+
+    def __init__(self, data, name, icon, locale):
+        """Initialize."""
+        self._data = data
+        self._icon = icon
+        self._locale = locale
+        self._name = name
+        self._state = None
+        self._unit = None
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._data:
+            return {
+                ATTR_ATTRIBUTION: 'AirVisual©',
+                ATTR_CITY: self._data.city,
+                ATTR_COUNTRY: self._data.country,
+                ATTR_STATE: self._data.state,
+                ATTR_TIMESTAMP: self._data.pollution_info.get('ts')
+            }
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return self._icon
+
+    @property
+    def name(self):
+        """Return the name."""
+        return '{0} {1}'.format(SENSOR_LOCALES[self._locale], self._name)
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self._state
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Update the status of the sensor."""
+        _LOGGER.debug('updating sensor: %s', self._name)
+        self._data.update()
+
+
+class AirPollutionLevelSensor(AirVisualBaseSensor):
+    """Define a sensor to measure air pollution level."""
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Update the status of the sensor."""
+        yield from super().async_update()
+        aqi = self._data.pollution_info.get('aqi{0}'.format(self._locale))
+
+        try:
+            [level] = [
+                i for i in POLLUTANT_LEVEL_MAPPING
+                if i['minimum'] <= aqi <= i['maximum']
+            ]
+            self._state = level.get('label')
+        except ValueError:
+            self._state = None
+
+
+class AirQualityIndexSensor(AirVisualBaseSensor):
+    """Define a sensor to measure AQI."""
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Update the status of the sensor."""
+        yield from super().async_update()
+        self._state = self._data.pollution_info.get(
+            'aqi{0}'.format(self._locale))
+
+
+class MainPollutantSensor(AirVisualBaseSensor):
+    """Define a sensor to the main pollutant of an area."""
+
+    def __init__(self, data, name, icon, locale):
+        """Initialize."""
+        super().__init__(data, name, icon, locale)
+        self._symbol = None
+        self._unit = None
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._data:
+            return merge_two_dicts(super().device_state_attributes, {
+                ATTR_POLLUTANT_SYMBOL: self._symbol,
+                ATTR_POLLUTANT_UNIT: self._unit
```

Comment:
Don't you want to have a unit of measure for these sensors? Why not overwrite the `unit_of_measurement` property?

Reply:
I see you're working on that now. :wink:

Reply:
Would love your thoughts here:

* The Air Quality Index doesn't really have a unit; I want to add _something_ so I get a nice graph, but not sure what it would be.
* The Air Quality Level doesn't have a unit, since its values are strings like "Good".
* Same for the Main Pollutant.

Reply:
Maybe use `'PSI'` as unit for the pollution index?

Reply:
`PSI` isn't quite right, as we're not specifically measuring pressure. [Sadly, the EPA specifically states that AQI](https://www.epa.gov/sites/production/files/2014-05/documents/zell-aqi.pdf):

>Ranges from 0 to 500 *(no units)*

Check out my latest push; not sure if you'll approve, but if I return an empty string, I get the bar chart that I want _and_ the `unit_of_measurement` attribute is the technically correct value: nothing.

Reply:
I was referring to pollutant standard index:
https://en.m.wikipedia.org/wiki/Pollutant_Standards_Index

An empty string could be OK, if we make that standard for all unit less measurements that should have a line chart.

Reply:
👍